FlexConfirmMail/Outlook
# Add support to specify trusted addresses

Add support to specify trusted addresses.

If a trusted domain entry contains "@", we regard it as address, not domain.

## Test

* Remove files in `%APPDATA%\FlexConfirmMail` if exist.
* Open Outlook and open setting dialog for FlexConfirmMail
* Open the "社内ドメイン" tab of the setting dialog.
  * Confirm that the comment have description for how to specify addresses like below  ...OK
    ```
    # (5) ドメインだけでなくアドレス全体を指定することもできます。
    #     * @を含む場合はアドレスが指定されたとみなします。
    #     * @を含まない場合はドメインが指定されたとみなします。
    #
    ```
* Add entries as following to the "社内ドメイン" setting
  ```
  test@a.co.jp
  *test@b.co.jp
  -*test@c.co.jp
  test2@*
  test.co.jp
  ```
* Try to send mail to `test@a.co.jp` and  `test3@a.co.jp` （test for `test@a.co.jp`）
  * Confirm that `test@a.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test3@a.co.jp` is in "社外の送信先" ...OK
* Try to send mail to `test-test@a.co.jp` and  `test-test@b.co.jp` (test for `*test@b.co.jp`)
  * Confirm that `test-test@b.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test-test@a.co.jp` is in "社外の送信先" ...OK
* Try to send mail to `test-test@a.co.jp` and `test-test@c.co.jp` (test for `-*test@c.co.jp`)
  * Confirm that `test-test@a.co.jp` and  `test-test@c.co.jp` are in "社外の送信先"  ...OK
* Try to send mail to `test2@a.co.jp` and `test2@c.co.jp` (test for `test2@*`)
  * Confirm that `test2@a.co.jp` and  `test2@c.co.jp` are in "社内の送信先" ...OK
  * Confirm that they are grouping each by  `a.co.jp` and `b.co.jp` ...OK
* Try to send mail to `test-test@test.co.jp` and `test-test@a.co.jp` (test for `test.co.jp`)
  * Confirm that `test-test@test.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test-test@a.co.jp` is in "社外の送信先" ...OK
